src/python/dxpy/sugar/__init__.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (C) 2013-2019 DNAnexus, Inc.
+#
+# This file is part of dx-toolkit (DNAnexus platform client libraries).
+#
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#   use this file except in compliance with the License. You may obtain a copy
+#   of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+from __future__ import print_function, unicode_literals, division, absolute_import
+from functools import wraps
+import re
+
+import dxpy
+
+from dxpy.sugar.processing import run_cmd, chain_cmds
+from dxpy.sugar.transfers import (
+    Uploader,
+    Downloader,
+    upload_file,
+    tar_and_upload_files,
+    download_file,
+)
+
+
+MEM_RE = re.compile(r"^MemTotal:[\s]*([0-9]*) kB")
```

Comment:
From the definitions and naming in `available_memory` I get the feeling you don't want the memory on the worker, but you want the free (unused) memory on the worker.

```suggestion
MEM_RE = re.compile(r"^MemFree:[\s]*([0-9]*) kB")
```

I'll add that I don't see this variable being used anywhere else. I think it's better to just write the regular expression in the `available_memory` function.

Reply:
I have a habit of defining regular expressions as constants to "cache" them, i.e. if available_memory is called in a loop, the re is only compiled once.

Reply:
Actually I think MemAvailable is the right one https://superuser.com/questions/980820/what-is-the-difference-between-memfree-and-memavailable-in-proc-meminfo

Reply:
Good catch